recipes/sdl/all/conanfile.py

```diff
@@ -247,18 +249,84 @@ def _configure_cmake(self):
+                    self._cmake.definitions["SDL_PULSEAUDIO_SHARED"] = self.deps_cpp_info["pulseaudio"].shared
+                self._cmake.definitions["SDL_SNDIO"] = self.options.sndio
+                if self.options.sndio:
+                    self._cmake.definitions["SDL_SNDIO_SHARED"] = self.deps_cpp_info["sndio"].shared
+                self._cmake.definitions["SDL_NAS"] = self.options.nas
+                if self.options.nas:
+                    cmake_extra_ldflags += ["-lXau"]  # FIXME: SDL sources doesn't take into account transitive dependencies
+                    cmake_required_includes += [os.path.join(self.deps_cpp_info["nas"].rootpath, str(it)) for it in self.deps_cpp_info["nas"].includedirs]
+                    self._cmake.definitions["SDL_NAS_SHARED"] = self.options["nas"].shared
+                self._cmake.definitions["SDL_X11"] = self.options.x11
+                if self.options.x11:
+                    self._cmake.definitions["HAVE_XEXT_H"] = True
+                self._cmake.definitions["SDL_X11_XCURSOR"] = self.options.xcursor
+                if self.options.xcursor:
+                    self._cmake.definitions["HAVE_XCURSOR_H"] = True
+                self._cmake.definitions["SDL_X11_XINERAMA"] = self.options.xinerama
+                if self.options.xinerama:
+                    self._cmake.definitions["HAVE_XINERAMA_H"] = True
+                self._cmake.definitions["SDL_X11_XINPUT"] = self.options.xinput
+                if self.options.xinput:
+                    self._cmake.definitions["HAVE_XINPUT_H"] = True
+                self._cmake.definitions["SDL_X11_XRANDR"] = self.options.xrandr
+                if self.options.xrandr:
+                    self._cmake.definitions["HAVE_XRANDR_H"] = True
+                self._cmake.definitions["SDL_X11_XSCRNSAVER"] = self.options.xscrnsaver
+                if self.options.xscrnsaver:
+                    self._cmake.definitions["HAVE_XSS_H"] = True
+                self._cmake.definitions["SDL_X11_XSHAPE"] = self.options.xshape
+                if self.options.xshape:
+                    self._cmake.definitions["HAVE_XSHAPE_H"] = True
+                self._cmake.definitions["SDL_X11_XVM"] = self.options.xvm
+                if self.options.xvm:
+                    self._cmake.definitions["HAVE_XF86VM_H"] = True
+                self._cmake.definitions["SDL_WAYLAND"] = self.options.wayland
+                if self.options.wayland:
+                    # FIXME: Otherwise 2.0.16 links with system wayland (from egl/system requirement)
+                    cmake_extra_ldflags += ["-L{}".format(os.path.join(self.deps_cpp_info["wayland"].rootpath, it)) for it in self.deps_cpp_info["wayland"].libdirs]
+                    self._cmake.definitions["SDL_WAYLAND_SHARED"] = self.options["wayland"].shared
+                    self._cmake.definitions["WAYLAND_SCANNER_1_15_FOUND"] = tools.Version(self.deps_cpp_info["wayland"].version) >= "1.15.0"
```

Comment:
```suggestion
                    self._cmake.definitions["WAYLAND_SCANNER_1_15_FOUND"] = 1  # FIXME: Check actual build-requires version
```

Probably here it also want to check the version from the _build_ context.